enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,288 @@
+
+This enhancement describes the flow for installing Single Node OpenShift using liveCD that perform the bootstrap logic and reboots to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Resource constraints are the primary motivation for single-node OCP clusters, and removing the need for a
+separate bootstrap machine significantly reduces the resources required to install the cluster.
+
+Requiring the auxiliary node for installing a Single Node OpenShift brings to play some requirements:
+1. The obvious additional node.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node OpenShift
+
+### Goals
+
+* This enhancement describes an approach for installing Single Node OpenShift in a BareMetal environment
+  for production use.
+* Minimal changes to openShift installer, this implementation shouldn't effect existing deployment flows.
+* Installation should result a clean Single Node OpenShift, without bootstrap leftovers.
+* Rely on a livecd, so it would be able to trigger the full installation using virtual media.
+* Self-contained bootstrap, The bootstrap flow is expressed as a single Ignition config which is layered on top of a pristine RHCOS image. It's important that we limit our use of this pattern to just the Ignition configs - as opposed to including other assets directly in the ISO image. 
+
+### Non-Goals
+
+* This enhancement does not address similar installation flow for 3 nodes cluster.
+* This enhancement does not address high-availability for single-node deployments.
+* This enhancement does not address single-node-developer cluster-profile installation.
+* This enhancement does not address cloud deployment for bootstrap in place (since livecd cannot be used). It will be addressed as part of future enhancement.
+* This enhancement does not address updating single-node clusters. It will be addressed as part of future enhancement.
+
+## Proposal
+
+Execute the bootstrap flow on a liveCD.
+Enrich the master Ignition with the control plane static pods and write this Ignition.
+Write the updated master Ignition + RHCOS to disk and reboot.
+Post reboot the node will complete the installation.
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+Ignition configuration.
+The user will boot RHCOS liveCD with this Ignition to create a single-node deployment.
+
+### Implementation Details/Notes/Constraints
+
+The OpenShift installer `create ignition-configs` command will generate a `bootstrap-in-place-for-live-iso.ign`
```

Comment:
Why do we need a separate ignition config file? Why wouldn't we continue to use the `bootstrap.ign` file?

Reply:
It's related to this thread: https://github.com/openshift/enhancements/pull/565#discussion_r543619003

Basically there needs to be a way to inject other [per-machine configuration](https://github.com/openshift/machine-config-operator/issues/1720) into this process.  One model for that is to do the same thing we're doing for UPI bare metal - customize the pointer config and then have that custom config include the `openshift-install` generated one.

But indeed if in that thread we add an environment variable or some sort of argument, then we'd have just one Ignition config that had everything done.

Reply:
@eranco74 suggested that we keep both bootstrap.ign and the bootstrap-in-place-for-live-iso.ign. In this way the default flow will not break (master=1 for crc for example) and only those who'll be interested in suing livecd, will use the new one. In this way, we'll not have to add another ENV variable to mark the installer which path to go.
Thoughts?

Reply:
I'd prefer if we can avoid creating unnecessary assets. I can see customers getting confused about this new asset and what they need to do with it. Can we use a separate target instead? e.g. `openshift-install create single-node-ignition-config` Do we still need to output master.ign and worker.ign under this target?

Reply:
I like the idea of a new target (better than env).
I think that `create single-node-ignition-config` should produce just the bootstrap-in-place.ign

Reply:
When this section is updated with the `create single-node-ignition-config` approach we should move the `create ignition-configs` approach to the alternatives section at the end of this doc so we have the history of why we changed.